src/RVRP_data.jl

```diff
@@ -1,118 +1,99 @@
-    location::Location
-    capacity_demand::Float64
-    time_windows::Vector{TimeWindow} # optional
-    duration::Float64 # optional
-    req_id::String
-end
-
 struct Operation <: AbstractNode
-    start_location::Location
-    end_location::Location
-    required_capacity::Float64
-    delta_capacity::Float64
+    location::Location
+    resource_consumptions::Dict{RESOURCE,Float64}
     time_windows::Vector{TimeWindow} # optional
-    duration::Float64 # optional      
-    req_id::String
+    req_index::Int
+    operation_type::OPERATION # Pickup, Delivery, Cleaning, ...
 end
 
-abstract type AbstractRequest end
-
-struct PickupRequest <: AbstractRequest
-    id::String
-    node::Pickup
-end    
-
-struct DeliveryRequest <: AbstractRequest
+struct SimpleRequest
     id::String
-    node::Delivery
-end    
-
-struct OperationRequest <: AbstractRequest
-    id::String
-    node::Operation
-end    
+    index::Int
+    operation::Operation # Single Requests have only one operation
+    request_type::REQUEST # SingleOperation, Shipment, ...
+end
 
-struct Shipment <: AbstractRequest
+struct ComplexRequest
     id::String
-    pickup::Pickup
-    delivery::Delivery
+    index::Int
+    operations::Vector{Operation} # Sequence of operations ; can be limited to one; can be a pair of Pickup and Delivery, or a triplets including a cleaning first, etc
+    request_type::REQUEST # SingleOperation, Shipment, ...
 end
 
-abstract type AbstractVehicle end
+struct VehicleBaseType
+    fixed_cost::Float64
+    resource_unit_prices::Vector{Float64}
+    resource_capacities::Vector{Float64}
+end
 
-struct Vehicle <: AbstractVehicle
+struct VehicleType
     id::String
+    index::Int
     depot::Depot # optional
-    v_type::VehicleType
+    base_type::VehicleType
     time_schedule::TimeWindow
     return_to_depot::Bool
     infinite_copies::Bool
-    initial_load::Float64
-    picked_shipments::Vector{Shipment}
+    nb_of_copies::Int
+    resource_intial_states::Vector{Float64}
+    ongoing_requests::Vector{Pair{Request,Int}}
 end
 
 struct RvrpProblem
     problem_id::String
     problem_type::ProblemType
-    vehicles::Vector{Vehicle}
-    vehicle_types::Vector{VehicleType}
+    vehicles::Vector{VehicleType}
     distance_matrix::Array{Float64,2}
     travel_times_matrix::Array{Float64,2}
+
     # Requests
-    pickups::Vector{PickupRequest}
-    deliveries::Vector{DeliveryRequest}
-    operations::Vector{OperationRequest}
-    shipments::Vector{Shipment}
-    picked_shipments::Vector{Shipment}
-end 
+    # Requests with only one operation of type Pickup:
+    pickups::Vector{SimpleRequest}
+
+    # Requests with only one operation of type Delivery:
+    deliveries::Vector{SimpleRequest}
+
+    # Requests with arbitrary number of operations of any OPERATION
+    # that must be done in the given sequence, by the same vehicle and
+    # preemption (between operations) is not allowed:
+    operations::Vector{ConplexRequest}
```

Comment:
This confusing. Operation is already used for an atomic operation within a Request. One of the two should be called differently

Reply:
Maybe the best thing is to remove this vector from this layer as they can be transformed to atomic operations (that we can add as deliveries) and using asymetric matrices for distance, time...

Reply:
I agree